src/widgets/DataTable/DataTableRow.js

```diff
@@ -42,187 +42,197 @@ import { formatStatus } from '../../utilities/index';
-                textInputStyle={cellText[cellAlignment]}
-                textStyle={editableCellUnfocused[cellAlignment]}
-                cellTextStyle={editableCellText}
-                rowIndex={rowIndex}
-              />
-            );
+    // Callback for rendering a row of cells.
+    const renderCells = useCallback(
+      () =>
+        // Map each column to an appropriate cell for a given row.
+        columns.map(({ key: columnKey, type, width, alignText }, index) => {
+          // Indicator if the right hand border should be removed from styles for this cell.
+          const isLastCell = index === columns.length - 1;
 
-          case 'checkable':
-            return (
-              <CheckableCell
-                key={columnKey}
-                rowKey={rowKey}
-                columnKey={columnKey}
-                isChecked={rowState && rowState.isSelected}
-                isDisabled={isDisabled}
-                CheckedComponent={CheckedComponent}
-                UncheckedComponent={UncheckedComponent}
-                DisabledCheckedComponent={DisabledCheckedComponent}
-                DisabledUncheckedComponent={DisabledUncheckedComponent}
-                onCheckAction={getAction(columnKey, 'onCheckAction')}
-                onUncheckAction={getAction(columnKey, 'onUncheckAction')}
-                dispatch={dispatch}
-                containerStyle={touchableCellContainer}
-                width={width}
-                isLastCell={isLastCell}
-              />
-            );
+          // This cell is disabled if the pageObject is finalised, the row has been explicitly set
+          // as disabled, or the rowData is disabled (i.e. data is an invoice),
+          const isDisabled =
+            isFinalised || (rowState && rowState.isDisabled) || rowData.isFinalised;
 
-          case 'editableDate':
-            return (
-              <NewExpiryDateInput
-                key={columnKey}
-                value={rowData[columnKey]}
-                rowKey={rowKey}
-                columnKey={columnKey}
-                editAction={getAction(columnKey)}
-                isFocused={isFocused}
-                isDisabled={isDisabled}
-                dispatch={dispatch}
-                width={width}
-                isLastCell={isLastCell}
-                rowIndex={rowIndex}
-              />
-            );
+          // Alignment of this particular column. Default to left hand ide.
+          const cellAlignment = alignText || 'left';
 
-          case 'status':
-            return (
-              <Cell
-                key={columnKey}
-                value={formatStatus(rowData[columnKey])}
-                width={width}
-                viewStyle={cellContainer[cellAlignment]}
-                textStyle={cellText[cellAlignment]}
-                isLastCell={isLastCell}
-              />
-            );
+          switch (type) {
+            case 'editableString':
+            case 'editableNumeric':
```

Comment:
I think the types have been updated to not use `'editable'`? Should be switching on `editable`?

Reply:
bit annoying - the `editable` field is mainly to be able to count the number of `editable` columns for easier focusing. The `type` is for easier switching for `cells` ...